src/components/Form/formFunctions.ts

```diff
@@ -6,9 +6,9 @@ import {
 import { Database } from "@/databaseTypesFile";
 import dayjs, { Dayjs } from "dayjs";
 
-type Field = Fields[keyof Fields];
-export type ErrorSetter = (errorKey: string, errorType: Errors) => void;
-export type FieldSetter = (fieldKey: string, newFieldValue: Field) => void;
+export type Setter<SpecificFields extends Fields> = <Key extends keyof SpecificFields>(
+    keyValuePairs: [Key, any][]
```

Comment:
I think a list of { key: Key, value: any } is easier read than a list of [Key, value] because we can name the two fields.
```suggestion
    keyValuePairs: { key: Key, value: any }[]
```


Or an even better way of doing this might be something like this, where the setter expects one object as a parameter, which can then set multiple fields at the same time?
```ts
export type Setter<SpecificFields extends Fields> = (keyValuePairs: {
    [key in keyof SpecificFields]: any;
}) => void;
```

Reply:
I agree that {key: Key, value: any}[] would be better, but I'm not sure I understand your second suggestion. If we were to just pass in one object, how would we then set multiple fields?

Reply:
Pushing the first suggestion :)

Reply:
If we pass in an object like 
```ts
{
  field1: value1,
  field2: value2,
}
```
then field1 should be updated to value1, field2 should be updated to value2 etc. Am I missing something?

Reply:
we got this to work